numpy/lib/_iotools.py

```diff
@@ -677,7 +682,22 @@ def _loose_call(self, value):
 
     def _strict_call(self, value):
         try:
-            return self.func(value)
+
+            # We check if we can convert the value using the current function
+            new_value = self.func(value)
+
+            # In addition to having to check whether func can convert the
+            # value, we also have to make sure that we don't get overflow 
+            # errors for integers.
+            if self.func is int:
+                try:
+                    np.array([value], dtype=self.type)
```

Comment:
It doesn't really matter, but you don't need to create a list with the value in it, `np.array(value, dtype=self.type)` will create a valid 0-dimensional array.